packages/python/plotly/plotly/express/_core.py

```diff
@@ -1482,7 +1607,13 @@ def build_dataframe(args, constructor):
         if not wide_mode and (no_x != no_y):
             for ax in ["x", "y"]:
                 if args.get(ax) is None:
-                    args[ax] = df_input.index if df_provided else Range()
+                    args[ax] = (
+                        index
+                        if index is not None
+                        else Range(
+                            label=_escape_col_name(columns, ax, [var_name, value_name])
+                        )
```

Comment:
@emilykl I noticed that in the performance benchmarking `px.funnel` was problematic, this change should solve the issue. Maybe take a double look at it 🙏🏼

Reply:
@FBruzzesi I can confirm the funnel chart is working now! I will take a closer look at this code.

I am seeing much improved performance for the pie chart as well.